### PR TITLE
Uniform TryGetValue usage

### PR DESCRIPTION
**Pull request by ncave**
Making `TryGetValue` usage more uniform.

### CODE REVIEW

File: src/absil/ilwrite.fs [hunk 1]
```diff
@@ -464,10 +464,9 @@ type MetadataTable<'T> =
 #if DEBUG
         tbl.lookups <- tbl.lookups + 1 
 #endif
-        let mutable res = Unchecked.defaultof<_>
-        let ok = tbl.dict.TryGetValue(x, &res)
-        if ok then res
-        else tbl.AddSharedEntry x
+        match tbl.dict.TryGetValue(x) with
```
Comment (forki) [inline src/absil/ilwrite.fs]: Please remove the parens

Comment (ncave) [reply src/absil/ilwrite.fs]: @forki Done.

File: src/utils/HashMultiMap.fs [hunk 2]
```diff
@@ -105,16 +103,17 @@ type internal HashMultiMap<'Key,'Value>(n: int, hasheq: IEqualityComparer<'Key>)
...
 
     member x.Replace(y,z) = 
         firstEntries.[y] <- z
 
     member x.TryFind(y) =
-        let mutable res = Unchecked.defaultof<'Value>
-        let ok = firstEntries.TryGetValue(y,&res)
-        if ok then Some(res) else None
+        match firstEntries.TryGetValue y with
+        | true, res -> Some(res)
```
[the remark this answers is not in the document]
Comment (ncave) [reply src/utils/HashMultiMap.fs]: @KevinRansom That's how it was before, fixed.

[the remark this answers is not in the document]
Comment (KevinRansom) [reply src/utils/HashMultiMap.fs]: @ncave, yeah I get it, it's the worst part of cleanup PR's.  You opt in to fixing all of the rubbish that was there befor :-)

**Comment by KevinRansom** [comment]
Thanks for this.

**Pull request merged**